hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/utils/db/managed/ManagedObject.java

```diff
@@ -27,8 +27,11 @@
 class ManagedObject<T extends RocksObject> implements AutoCloseable {
   private final T original;
 
+  private final StackTraceElement[] elements;
+
   ManagedObject(T original) {
     this.original = original;
+    this.elements = Thread.currentThread().getStackTrace();
```

Comment:
I'm not sure it's a good idea to pay the cost of capturing the current stacktrace for every allocated RockObject in production. Maybe we should do that for debugging purpose only (when `LOG.isDebugEnabled()`).

Reply:
Good point.  If stack trace is used only when debug is enabled, we can limit its creation, too.